lib/bolt/transport/sudoable/connection.rb

```diff
@@ -78,7 +78,11 @@ def execute(*_args)
         # for task input data because the sudo password has already either been
         # provided on stdin or was not needed.
         def prepend_sudo_success(sudo_id, command_str)
-          "sh -c 'echo #{sudo_id} 1>&2; #{command_str}'"
+          "sh -c 'echo #{sudo_id} 1>&2; cd && #{command_str}'"
+        end
+
+        def prepend_chdir(command_str)
+          "sh -c 'cd && #{command_str}'"
```

Comment:
This will change the way we interpret commands, by now executing them in a shell. I'm not sure if that's desirable or not...

Reply:
I'm not happy with this change. But I also don't really see how to accomplish this another way...

Reply:
Oh this is how this already works. Perfect.

Reply:
This does slightly change behavior - one side effect is that `&&` is respected now over SSH when using run as. A few examples of how behavior differs now:
```
lucy ~/githubs/bolt ((8e632fb...)) $ bolt command run "whoami && whoami" --run-as ubuntu --sudo-password puppet -t thirsty-soap.delivery.puppetlabs.net
CLI arguments ["run-as", "sudo-password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
Finished on thirsty-soap.delivery.puppetlabs.net:
  STDOUT:
    ubuntu
    root
Successful on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 0.57 sec

lucy ~/githubs/bolt ((8e632fb...)) $ be bolt command run "whoami && whoami" --run-as ubuntu --sudo-password puppet -t thirsty-soap.delivery.puppetlabs.net
CLI arguments ["run-as", "sudo-password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
Finished on thirsty-soap.delivery.puppetlabs.net:
  STDOUT:
    ubuntu
    ubuntu
Successful on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 0.57 sec

lucy ~/githubs/bolt ((8e632fb...)) $ be bolt command run "cd && pwd" --run-as ubuntu --sudo-password puppet -t thirsty-soap.delivery.puppetlabs.net
CLI arguments ["run-as", "sudo-password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
Finished on thirsty-soap.delivery.puppetlabs.net:
  STDOUT:
    /home/ubuntu
Successful on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 0.62 sec

lucy ~/githubs/bolt ((8e632fb...)) $ bolt command run "cd && pwd" --run-as ubuntu --sudo-password puppet -t thirsty-soap.delivery.puppetlabs.net
CLI arguments ["run-as", "sudo-password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
Failed on thirsty-soap.delivery.puppetlabs.net:
  The command failed with exit code 1
  STDERR:
    sudo: cd: command not found
Failed on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 0.56 sec
```

Reply:
I think supporting `&&` is actually generally desirable since people are surprised it doesn't work. I wonder if supporting it just for `run-as` use cases will be annoying and confusing though, since it will work sometimes and not others with no explanation:
```
lucy ~/githubs/bolt ((8e632fb...)) $ be bolt command run "cd && pwd" --run-as ubuntu --sudo-password puppet -t thirsty-soap.delivery.puppetlabs.net
CLI arguments ["run-as", "sudo-password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
Finished on thirsty-soap.delivery.puppetlabs.net:
  STDOUT:
    /home/ubuntu
Successful on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 0.59 sec

lucy ~/githubs/bolt ((8e632fb...)) $ be bolt command run "cd && pwd" -u ubuntu -p password -t thirsty-soap.delivery.puppetlabs.netCLI arguments ["user", "password"] may be overridden by Inventory: /home/lucy/githubs/bolt/inventory.yaml
Started on thirsty-soap.delivery.puppetlabs.net...
all authorization methods failed (tried none, publickey, password)
Failed on thirsty-soap.delivery.puppetlabs.net:
  Authentication failed for user ubuntu@thirsty-soap.delivery.puppetlabs.net
Failed on 1 target: thirsty-soap.delivery.puppetlabs.net
Ran on 1 target in 2.06 sec
```

Reply:
It would _seem_ that we already support `&&` when not using run-as, so I think this is _just_ a bug fix in the run-as case and otherwise not a behavior change?